File: Java/SpaceVIL/spacevil/src/main/java/com/spvessel/spacevil/Core/DropArgs.java
package com.spvessel.spacevil.Core;

import java.util.List;

import com.spvessel.spacevil.Prototype;

/**
 * A class that describe mouse "drag and drop" input (file system).
 */
public final class DropArgs extends InputEventArgs {
    /**
     * An item on which the drop event occurs.
     */
    public Prototype item = null;

    /**
     * Number of selected paths.
     */
    public int count = -1;

    /**
     * List of selected paths.
     */
    public List<String> paths = null;

    /**
     * Clearing DropArgs.
     */
    @Override
    public void clear() {
        count = -1;
        paths = null;
        item = null;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append(item.getItemName() + "\n");
        sb.append(count + " \n");
        if (paths != null)
        {
            for (String p : paths)
            {
                sb.append(p + "\n");
            }
        }
        return sb.toString(); 
    }
}

File: CSharp/SpaceVIL/SpaceVIL/Core/DropArgs.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SpaceVIL.Core
{
    /// <summary>
    /// A class that describe mouse "drag and drop" input (file system).
    /// </summary>
    public sealed class DropArgs : IInputEventArgs
    {
        /// <summary>
        /// An item on which the drop event occurs.
        /// </summary>
        public Prototype Item = null;

        /// <summary>
        /// Number of selected paths.
        /// </summary>
        public int Count = -1;

        /// <summary>
        /// List of selected paths.
        /// </summary>
        public List<String> Paths = null;
        
        /// <summary>
        /// Clearing DropArgs.
        /// </summary>
        public void Clear()
        {
            Count = -1;
            Paths = null;
            Item = null;
        }

        public override String ToString()
        {
            StringBuilder sb = new StringBuilder();
            sb.AppendLine(Item.GetItemName());
            sb.AppendLine(Count + " ");
            if (Paths != null)
            {
                foreach (var p in Paths)
                {
                    sb.AppendLine(p);
                }
            }

            return sb.ToString();
        }
    }
}